### PR TITLE
Changes to adapt mod files for MOD to CPP translation

### PR DESCRIPTION
**Pull request by pramodk**
* Remove random123 APIs as they are now included via headers
* Avoid passing void*
* Avoid assignement of void*

See https://github.com/neuronsimulator/nrn/pull/1597

### CODE REVIEW

File: mod/invlfire.mod [hunk 1]
```diff
@@ -161,7 +147,7 @@ VERBATIM
   if (ifarg(1)) {
     *pv = nrn_random_arg(1);
   }else{
-    *pv = (void*)0;
+    *pv = nullptr;
```
Comment (olupton) [inline mod/invlfire.mod]:
```suggestion
    *pv = (Rand*)0;
```
if we want to remain compatible with C / older NEURON versions

File: mod/invlfire.mod [hunk 2]
```diff
@@ -180,7 +166,7 @@ VERBATIM
   _ran_compat = 2;
   if (*pv) {
     nrnran123_deletestream(*pv);
-    *pv = (nrnran123_State*)0;
+    *pv = nullptr;
```
Comment (olupton) [inline mod/invlfire.mod]: As above; need to decide if we're supporting C / old NEURON.

File: mod/invlfire.mod [hunk 3]
```diff
@@ -223,7 +209,7 @@ static void bbcore_write(double* x, int* d, int* xx, int *offset, _threadargspro
 #if NRNBBCORE
       /* CORENeuron does not call DESTRUCTOR so... */
       nrnran123_deletestream(*pv);
-      *pv = (nrnran123_State*)0;
+      *pv = nullptr;
```
Comment (olupton) [inline mod/invlfire.mod]: As above; need to decide if we're supporting C / old NEURON.

File: modx/invlfiresha.mod [hunk 4]
```diff
@@ -217,7 +203,7 @@ VERBATIM
   if (ifarg(1)) {
     *pv = nrn_random_arg(1);
   }else{
-    *pv = (void*)0;
+    *pv = nullptr;
```
Comment (olupton) [inline modx/invlfiresha.mod]: As above; need to decide if we're supporting C / old NEURON.

File: modx/invlfiresha.mod [hunk 5]
```diff
@@ -300,7 +286,7 @@ z = 1;
 #if NRNBBCORE
       /* CORENeuron does not call DESTRUCTOR so... */
       nrnran123_deletestream(*pv);
-      *pv = (nrnran123_State*)0;
+      *pv = nullptr;
```
Comment (olupton) [inline modx/invlfiresha.mod]: As above; need to decide if we're supporting C / old NEURON.

**Review by olupton — COMMENTED** [review]
Generally looks good, all my comments are basically the same.
`nullptr` is not going to work with the current `master` of NEURON.
I **tend** to think we will make our lives easier if we _initially_ make these external models/tests dual C/C++ compatible, but I am open to other options.

**Review by nrnhines — APPROVED** [review]

**Comment by pramodk** [comment]
> I tend to think we will make our lives easier if we initially make these external models/tests dual C/C++ compatible, but I am open to other options.

In this case it's trivial do doesn't hurt. Done. (But in near future we might need bite the bulltet)

**Review by olupton — APPROVED** [review]

**Comment by olupton** [comment]
Before we merge this we should test the `master` of NEURON against the various updated test/model repositories.

**Comment by pramodk** [comment]
> Before we merge this we should test the master of NEURON against the various updated test/model repositories.

Will create PR where we can gather these updated masters.

**Pull request merged**